### DISCUSSION TITLE
Is it possible to run terminal commands when uninstalling the executable?

### DISCUSSION THREAD

**Discussion by matteocollina** [tauri-apps/tauri#5888] [Q&A]
I, when the user **uninstalls the application,** would like to perform certain actions either via terminal commands or js commands.

**Comment by FabianLars** [comment]
This is in general a pretty tricky subject, primarily because of platform and package differences.

For example on macOS there is no classic uninstallation step/event for `.app` and `.dmg` bundles afaik. `.pkg` (not directly supported by tauri) only has postinstall scripts if i remember it correctly.

On Linux (only considering what Tauri supports right now):
- AppImages: no install/uninstall step, the user just deletes the file
- `.deb`: This one indeed has a postrm script, but apparently everyone discourages its use, example: https://www.debian.org/doc/manuals/maint-guide/dother.en.html#maintscripts

Windows:
- Obviously if you distribute just the .exe (which we don't recommend because of the missing webview installation, and missing feature) you'll have the same problem as the AppImage.
- .msi, WiX kindaaa supports this, so you could [modify tauri's default wix template](https://tauri.app/v1/guides/building/windows#replacing-the-installer-code-with-a-custom-wix-file). some resources: https://stackoverflow.com/questions/320921/how-to-add-a-wix-custom-action-that-happens-only-on-uninstall-via-msi/51215272#51215272 / https://developernote.com/2018/03/a-simple-wix-installer-that-runs-custom-actions-on-install-and-uninstall/ - keep in mind that your actual tauri app will already be removed at that point (meaning _no_ js commands or anything from your tauri app). Also from what i've seen so far it's super annoying to work with in general 😅 
- NSIS installer-exe coming in v1.3: I think NSIS should also support it, but v1.3 won't have an option to modify the built-in nsis template yet.

**Reply by yk0n9** [reply]
So, is it possible to execute custom NSI scripts now?

**Reply by FabianLars** [reply]
> but v1.3 won't have an option to modify the built-in nsis template yet.

-> No, but 1.4 will have it: https://github.com/tauri-apps/tauri/pull/6922
